apps/prairielearn/src/lib/regrading.js

```diff
@@ -6,100 +6,73 @@ const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'
 
 import { logger } from '@prairielearn/logger';
 import { createJob, createJobSequence, failJobSequence } from './server-jobs-legacy';
+import { createServerJob } from './server-jobs';
 import * as sqldb from '@prairielearn/postgres';
 import * as ltiOutcomes from './ltiOutcomes';
+import { z } from 'zod';
+import { IdSchema } from './db-types';
 
 const sql = sqldb.loadSqlEquiv(__filename);
 
-export function regradeAssessmentInstance(
-  assessment_instance_id,
-  user_id,
-  authn_user_id,
-  callback,
-) {
-  debug('regradeAssessmentInstance()');
-  var params = { assessment_instance_id };
-  sqldb.queryOneRow(sql.select_regrade_assessment_instance_info, params, function (err, result) {
-    if (ERR(err, callback)) return;
-    const assessment_instance_label = result.rows[0].assessment_instance_label;
-    let jobInfo;
-    if (result.rows[0].user_uid) {
-      jobInfo = result.rows[0].user_uid;
-    } else {
-      jobInfo = 'group name ' + result.rows[0].group_name;
-    }
-    const assessment_id = result.rows[0].assessment_id;
-    const course_instance_id = result.rows[0].course_instance_id;
-    const course_id = result.rows[0].course_id;
+export async function regradeAssessmentInstance(assessment_instance_id, user_id, authn_user_id) {
+  const assessmentInstance = await sqldb.queryRow(
+    sql.select_regrade_assessment_instance_info,
+    { assessment_instance_id },
+    z.object({
+      assessment_instance_label: z.string(),
+      user_uid: z.string().optional(),
+      group_name: z.string().nullable(),
+      assessment_id: IdSchema,
+      course_instance_id: IdSchema,
+      course_id: IdSchema,
+    }),
+  );
+  const assessment_instance_label = assessmentInstance.assessment_instance_label;
+  let jobInfo;
+  if (assessmentInstance.user_uid) {
+    jobInfo = assessmentInstance.user_uid;
+  } else {
+    jobInfo = 'group name ' + assessmentInstance.group_name;
+  }
+  const assessment_id = assessmentInstance.assessment_id;
+  const course_instance_id = assessmentInstance.course_instance_id;
+  const course_id = assessmentInstance.course_id;
 
-    var options = {
-      course_id: course_id,
-      course_instance_id: course_instance_id,
-      assessment_id: assessment_id,
-      user_id: user_id,
-      authn_user_id: authn_user_id,
-      type: 'regrade_assessment_instance',
-      description: 'Regrade ' + assessment_instance_label + ' for ' + jobInfo,
-    };
-    createJobSequence(options, function (err, job_sequence_id) {
-      if (ERR(err, callback)) return;
-      callback(null, job_sequence_id);
+  var options = {
+    courseId: course_id,
+    courseInstanceId: course_instance_id,
+    assessmentId: assessment_id,
+    userId: user_id,
+    authnUserId: authn_user_id,
+    type: 'regrade_assessment_instance',
+    description: 'Regrade ' + assessment_instance_label + ' for ' + jobInfo,
+  };
+  const serverJob = await createServerJob(options);
```

Comment:
Nit: consider inlining the `options` object into the call to `createServerJob`. There's also not really a reason to pick off `assessment_id`, `course_instance_id`, and `course_id` into local variables since they're only used once.

Reply:
Inlined `options` into `createServerJob` and removed local variables for `assessment_id`, `course_instance_id`, and `course_id` in commit [ecafed4](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/ecafed4ec68b0006158dd7f2866a9331734aa370).